lib/pbench/server/db/models/tracker.py

```diff
@@ -0,0 +1,546 @@
+        Integer, ForeignKey("dataset.id", ondelete="CASCADE"), nullable=False
+    )
+
+    def __init__(self, **kwargs):
+        """
+        Construct a new Metadata object for a dataset
+
+        Parameters:
+
+            key     Metadata key (from METADATA_KEYS)
+            value   String value for key
+        """
+        super().__init__(**kwargs)
+        try:
+            key = kwargs.get("key")
+        except KeyError:
+            raise MetadataBadKey("Metadata key is required")
+        else:
+            if key not in Metadata.METADATA_KEYS:
+                raise MetadataBadKey(f"Metadata key {key} is not supported")
+            self.key = key
+        try:
+            value = kwargs.get("value")
+        except KeyError:
+            raise MetadataBadKey(f"Metadata key {key} value is required")
+        else:
+            self.value = value
+
+    @staticmethod
+    def create(**kwargs):
+        try:
+            meta = Metadata(**kwargs)
+            meta.add(kwargs.get("dataset"))
+        except KeyError:
+            raise MetadataError("Metadata create missing argument 'dataset'")
+        except Exception:
+            return None
+        else:
+            return meta
+
+    @staticmethod
+    def get(dataset, key):
+        try:
+            meta = (
+                Database.db_session.query(Metadata)
+                .filter_by(dataset=dataset, key=key)
+                .first()
+            )
+        except SQLAlchemyError as e:
+            raise DatasetSqlError(f"Error getting {dataset} key {key}: {e}")
+        else:
+            if meta is None:
+                raise MetadataNotFound(f"No metadata {key} for {dataset}")
+            return meta
+
+    @staticmethod
+    def remove(dataset, key):
+        """
+        remove Remove a metadata key from the dataset
+
+        Args:
+            dataset (Dataset): Dataset with which key is associated
+            key (string): Name of metadata key to remove
+
+        Raises:
+            DatasetSqlError: Something went wrong
+        """
+        try:
+            Database.db_session.query(Metadata).filter_by(
+                dataset=dataset, key=key
+            ).delete()
+            Database.db_session.commit()
+        except SQLAlchemyError as e:
+            raise DatasetSqlError(f"Error deleting {dataset} key {key}: {e}")
+
+    def __str__(self):
+        return f"{self.dataset}>>{self.key}"
+
+    def add(self, dataset):
+        """
+        add Add the Metadata object to the dataset
+        """
+        try:
+            Metadata.get(dataset, self.key)
+        except MetadataNotFound:
+            pass
+        else:
+            raise MetadataDuplicateKey(f"{dataset} already has metadata key {self.key}")
+
+        try:
+            dataset.metadatas.append(self)
+            Database.db_session.add(self)
+            Database.db_session.commit()
+        except Exception as e:
+            Database.db_session.rollback()
```

Comment:
Should we also be undoing line 518 here?

Reply:
Yes!